ldap/servers/slapd/opshared.c

```diff
@@ -764,8 +765,10 @@ op_shared_search(Slapi_PBlock *pb, int send_result)
             }
         } else {
             /* be_suffix null means that we are searching the default backend
-             * -> don't change the search parameters in pblock */
-           if (be_suffix != NULL) {
+             * -> don't change the search parameters in pblock
+             * Also, we skip this block for 'cn=monitor' search and its subsearches
+             * as they are done by callbacks from monitor.c */
+            if ((be_suffix != NULL) && !slapi_sdn_issuffix(basesdn, monitorsdn)) {
```

Comment:
This logic may be clearer as:

```
if (!( be_suffix == NULL) || slapi_sdn_issuffix() ))
```

Reply:
Probably just a typo, but I think you meant  (!((be_suffix == NULL) || slapi_sdn_issuffix())) - the third parenthesis. 
But yeah, I agree, it sounds more cleaner:) Fixed